### DISCUSSION TITLE
[BP#33] Suppression de "Supprimer les balises images dont l’attribut SRC est vide"

### DISCUSSION THREAD

**Discussion by arnolem** [cnumr/best-practices#119] [Bonnes pratiques] [answered]
Supprimer la règle : https://github.com/cnumr/best-practices/blob/main/chapters/BP_033_fr.md
Je pense qu'il faut supprimer cette bonne pratique car :

 - J'ai testé rapidement sur plusieurs navigateurs et ça ne semble pas créer de problème :
 https://jsfiddle.net/cr3wen10/
  - Sur plus de 200 projets, je n'ai jamais rencontré de développeur qui créé des balises `<img>` avec un `src` vide (ou sans `src`).
  - En cherchant le pourquoi de cette règle, je ne vois que le coup du LazyLoading. Si c'est uniquement pour ce cas là, autant le préciser dans la règle #037 (si on prouve la véracité de cette règle)

**Comment by rlemaire** [comment]
Je suppose que ça a du arriver à l'époque de la V1.

Mais c'est plus une erreur de programmation à corriger qu'une optimisation en effet.

**Comment by tbroyer** [comment]
L'été dernier, [j'écrivais](https://groups.google.com/g/ecoconceptionweb/c/pNjmyPEpOEM/m/7ufcKcuzBwAJ):

> un src="", en plus d'être inutile comparé à l'absence de l'attribut, génère un événement 'onerror'; il y a quelques années, ça générait aussi une requête HTTP à la même URL que la page, dans le cas où elle fasse de la négociation de contenu, mais ça ne semble plus être le cas, en tous cas dans Chrome –et donc Edge et Opera– et Firefox

(note: pour l'événement onerror, la preuve ici: https://jsfiddle.net/pjmyn39b/)

Je suis d'accord pour supprimer cette BP, l'impact de ce onerror est négligeable comparé à l'impact d'une requête au serveur il y a quelques années.

**Comment by ACTLEM** [comment, marked as answer]
https://github.com/cnumr/best-practices/issues/131
